hail/python/hailtop/aiocloud/aioazure/client/network_client.py

```diff
@@ -5,7 +5,12 @@
 
 
 class AzureNetworkClient(AzureBaseClient):
-    def __init__(self, subscription_id, resource_group_name, session: Optional[AzureSession] = None):
-        session = session or AzureSession()
+    def __init__(self, subscription_id, resource_group_name, session: Optional[AzureSession] = None, **kwargs):
+        if 'params' not in kwargs:
+            kwargs['params'] = {}
+        params = kwargs['params']
```

Comment:
I'm a little confused here. Shouldn't params be per request?

Reply:
This sets the default params in the Session. You can override them with a request.